src/components/modals/WcSmartContractModal.tsx

```diff
@@ -185,32 +190,31 @@ async function wcRequestResponse(wallet: EdgeCurrencyWallet, uri: string, approv
   }
 }
 
-const asEvmPayloadMethod = asValue(
-  'personal_sign',
-  'eth_sign',
-  'eth_signTypedData',
-  'eth_signTypedData_v4',
-  'eth_sendTransaction',
-  'eth_signTransaction',
-  'eth_sendRawTransaction'
-)
-const asEvmWcRpcPayload = asObject({
+// [message, account, password]: personal_sign (password not supported yet)
+// [address, message]: eth_sign, eth_signTypedData, eth_signTypedData_v4
+
+const asEvmSignMethod = asValue('personal_sign', 'eth_sign', 'eth_signTypedData', 'eth_signTypedData_v4')
+const asEvmSignPayload = asObject({
   id: asEither(asString, asNumber),
-  method: asEvmPayloadMethod,
-  params: asEither(
-    asTuple(
-      asObject({
-        from: asString,
-        to: asOptional(asString),
-        data: asString,
-        gas: asOptional(asString),
-        gasPrice: asOptional(asString),
-        value: asOptional(asString),
-        nonce: asOptional(asString)
-      }),
-      asOptional(asString, '')
-    ),
-    asArray(asString)
+  method: asEvmSignMethod,
+  params: asEither(asTuple(asString, asString))
```

Comment:
This `asEither` isn't needed anymore, since there is only one option.

Also, it looks like personal_sign only uses params[0]. I'm not sure why the cleaner needs two parameters if we only care about the first one, but maybe there is a reason. Otherwise, `asTuple(asString, asOptional(asString))` or `asArray(asString)` might be needed, followed by some extra checking inside the methods themselves.